code-challenges.test.js

```diff
@@ -11,42 +11,130 @@
+    it("should return strings without vowels", () => {
+        expect(encodedString("xyz")).toBe("xyz")
+    })
+    it("should return strings with upper and lowercase vowels", () => {
+        expect(encodedString("CaT")).toBe("C4T")
+    })
+})
+    // b) Create the function that makes the test pass.
+        const encodedString = (string) => {
+            const vowels = {"a": "4", "e": "3", "i": "1", "o": "0"}
+        let encodedStr = ("")
+        for (let i = 0; i < string.length; i++) {
+            let char = string[i].toLowerCase();
+            if (vowels[char]) {
+              encodedStr += vowels[char];
+            } else {
+              encodedStr += string[i];
+            }
+          }
+          
+          return encodedStr;
+        }
 // --------------------2) Create a function that takes in an array of words and a single letter and returns an array of all the words containing that particular letter.
-
+        // Pseudocode:
+            // Given an array of fruits 
+            // Create 4 tests that describe different situations that the function should be able to test for
+            // Use the given variables to make sure the expected output is correct
+            // Create 2 test cases that define the variables 'letterA' and 'letterE' to filter the array
+            // Define our 'givenExpectedOutput' variable so it contains the expected output 
+            // Define the filterTheWords function that takes an array of strings array and a string letter as input parameters
+            // Use the "filter" method to create a new array that contains ONLY the string that "include" the specific letter that out outputs call for
+            // Return the filterTheWords function to return the filtered array
 // a) Create a test with expects statement using the variable provided.
 
-const fruitArray = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach", "Kiwi"]
-
-const letterA = "a"
-// Expected output: ["Mango", "Apricot", "Peach"]
-const letterE = "e"
-// Expected output: ["Cherry", "Blueberry", "Peach"]
-
-// b) Create the function that makes the test pass.
+describe("filterTheWords", () => {
+    const fruitArray = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach", "Kiwi"]
+    it("returns an array of words with the letter 'a' in it", () => {
+        const letterA = "a"
+        const givenExpectedOutput = ["Mango", "Apricot", "Peach"]
+        const ourResult = filterTheWords(fruitArray, letterA)
+        expect(ourResult).toEqual(givenExpectedOutput)
+    })
+   it("returns an array of words with the letter 'e' in it", () => {
+    const letterE = 'e'
+    const givenExpectedOutput = ["Cherry", "Blueberry", "Peach"]
+    const ourResult = filterTheWords(fruitArray, letterE)
+    expect(ourResult).toEqual(givenExpectedOutput)
+   })
+})
+    // b) Create the function that makes the test pass.
+    const filterTheWords = (array, letter) => {
+        return array.filter((word) => word.includes(letter))
+    }
```

Comment:
This is super clean and I love the way you went about this! The only issue is that some words will have the letter, but it will be upper case, which won't pass the condition you set!